modules/generic-scala3/src/main/scala/pureconfig/generic/semiauto.scala

```diff
@@ -12,6 +12,9 @@ object semiauto {
   export HintsAwareConfigWriterDerivation.deriveWriter
   export EnumDerivation._
 
+  inline def deriveConvert[A <: AnyVal]: ConfigConvert[A] =
```

Comment:
It's not redundant. We have a compile-time check, but only for the recursive case of `summonConfigReader` where we don't know anything about the `A`. 
Mirrors are not synthesized for value classes, so it's not possible to call the `inline def deriveReader[A](using m: Mirror.Of[A]): ConfigReader[A]` version on them directly. We need either use the overload with `[A <: AnyVal]` or change the signature to the `inline def deriveReader[A]: ConfigReader[A]` and do inside check similar to the `summonConfigReader`, but without the first case. 
I think the solution with overload is cleaner.

Reply:
After giving it some more thought, I think we might actually need to hide everything behind `inline def deriveReader[A]: ConfigReader[A]` anyway at some point, as it might become more cluttered with potential future opaque types derivation and auto derivation 🤔 .

I'll try experimenting with it a little bit more.

Reply:
In the end I exposed in under simplified interface.